modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/network.scala

```diff
@@ -0,0 +1,52 @@
+package org.tessellation.rosetta.server.model
+
+import cats.Show
+import cats.implicits.toBifunctorOps
+
+import derevo.cats.show
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.predicates.all.NonEmpty
+import eu.timepit.refined.refineV
+import eu.timepit.refined.types.string.NonEmptyString
+import io.estatico.newtype.macros.newtype
+import io.getquill.MappedEncoding
+
+object network {
+  implicit def showNonEmptyString: Show[NonEmptyString] = s => s.value
+
+  @derive(show)
+  @newtype
+  case class NetworkStatus(value: NonEmptyString)
+
+  object NetworkStatus {
+    val accepted: NetworkStatus = NetworkStatus("Accepted")
+    val reverted: NetworkStatus = NetworkStatus("Reverted")
+    val canceled: NetworkStatus = NetworkStatus("Canceled")
+
+    implicit val quillEncode: MappedEncoding[NetworkStatus, String] =
+      MappedEncoding[NetworkStatus, String](_.value.value)
+
+    implicit val quillDecode: MappedEncoding[String, NetworkStatus] = MappedEncoding[String, NetworkStatus](
+      refineV[NonEmpty].apply[String](_).leftMap(new Throwable(_)) match {
+        case Left(err)    => throw err
+        case Right(value) => NetworkStatus(value)
+      }
+    )
+
+    def fromString(value: String): Option[NetworkStatus] = {
+      val normalizedString = value.toLowerCase()
+
+      if (normalizedString == accepted.value.value.toLowerCase())
+        return Some(accepted)
+
+      if (normalizedString == reverted.value.value.toLowerCase())
+        return Some(reverted)
+
+      if (normalizedString == canceled.value.value.toLowerCase())
+        return Some(canceled)
+
+      None
+    }
```

Comment:
it can be handled by enum

Reply:
Reworked this to an enum, renamed the file `NetworkStatus.scala`, and relocated it into the `org.tessellation.rosetta.server.model.enums` package.

Reply:
I have one more suggestion - in the whole codebase, we don't group files and classes by type but by feature.
So packages shouldn't be:
- utils
- traits
- enums
and so on, but rather
- transactions
- blocks
- snapshots
- consensus
- gossip
- ...

That being said, you can group these enums by feature (for example `transaction.TransactionOperationStatus` or `search.Operator`

Reply:
The `/search` endpoints stuff in `org.tessellation.rosetta.server.model` have all been relocated to `org.tessellation.rosetta.server.model.search`. In `org.tessellation.rosetta.api.dag`, `BlockEventType` has been relocated under `org.tessellation.rosetta.api.dag.block`, `SignatureType` has been relocated under `org.tessellation.rosetta.api.dag.crypto`, and `ChainObjectStatus` and `NetworkChainObjectStatus` has been relocated under `org.tessellation.rosetta.api.dag.network`. I am not really happy about the `network` package name and welcome suggestions for a better name.